Algorithm.Python/CustomSecurityInitializerAlgorithm.py

```diff
@@ -0,0 +1,77 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from clr import AddReference
+AddReference("System")
+AddReference("QuantConnect.Algorithm")
+AddReference("QuantConnect.Common")
+
+from System import *
+from QuantConnect import *
+from QuantConnect.Algorithm import *
+from QuantConnect.Brokerages import *
+from QuantConnect.Data import BaseData
+from QuantConnect.Data.Market import *
+from QuantConnect.Securities import *
+
+### <summary>
+### This algorithm shows how to set a custom security initializer.
+### A security initializer is run immediately after a new security object
+### has been created and can be used to security models and other settings,
+### such as data normalization mode
+### </summary>
+### <meta name="tag" content="using data" />
+### <meta name="tag" content="securities and portfolio" />
+### <meta name="tag" content="trading and orders" />
+class CustomSecurityInitializerAlgorithm(QCAlgorithm):
+
+    def Initialize(self):
+        
+        # set our initializer to our custom type
+        self.SetBrokerageModel(BrokerageName.InteractiveBrokersBrokerage)
+        
+        func_security_seeder = FuncSecuritySeeder(Func[Security, BaseData](self.GetLastKnownPrice))
+        self.SetSecurityInitializer(CustomSecurityInitializer(self.BrokerageModel, func_security_seeder, DataNormalizationMode.Raw))
+        
+        self.SetStartDate(2013,10,01)
+        self.SetEndDate(2013,11,01)
+
+        self.AddEquity("SPY", Resolution.Hour)
+
+    def OnData(self, data):
+        if not self.Portfolio.Invested:
+            self.SetHoldings("SPY", 1)
+
+
+class CustomSecurityInitializer(BrokerageModelSecurityInitializer):
+    '''Our custom initializer that will set the data normalization mode.
+    We sub-class the BrokerageModelSecurityInitializer so we can also
+    take advantage of the default model/leverage setting behaviors'''
+
+    def __init__(self, brokerageModel, securitySeeder, dataNormalizationMode):
+        '''Initializes a new instance of the CustomSecurityInitializer class with the specified normalization mode
+        brokerageModel -- The brokerage model used to get fill/fee/slippage/settlement models
+        securitySeeder -- The security seeder to be used
+        dataNormalizationMode -- The desired data normalization mode'''
+        self.base = BrokerageModelSecurityInitializer(brokerageModel, securitySeeder)
+        self.dataNormalizationMode = dataNormalizationMode
+
+    def Initialize(self, security, seedSecurity):
+        '''Initializes the specified security by setting up the models
+        security -- The security to be initialized
+        seedSecurity -- True to seed the security, false otherwise'''
+        # first call the default implementation
+        self.base.Initialize(security, seedSecurity)
```

Comment:
I tried to subclass. In python it would be something like
`super(self.__class__, self).Initialize(security, seedSecurity)`
In __init__ I would have
`super(self.__class__, self).__init__(brokerageModel, securitySeeder)`
instead of
`self.base = BrokerageModelSecurityInitializer(brokerageModel, securitySeeder)`
but Initialize above wouldn't belong to the instance I have created.